generator/templates/rust/crate/src/lib.rs.mustache

```diff
@@ -137,23 +141,36 @@ impl {{NameToPascal}} {
         {{#QueryParams}}
         let builder = gax::query_parameter::add(builder, "{{JSONName}}", {{{AsQueryParameter}}}).map_err(Error::other)?;
         {{/QueryParams}}
-        let res = builder
-            .bearer_auth(&client.cred.access_token().await.map_err(Error::authentication)?.value)
-            {{#HasBody}}
-            .json(&req{{BodyAccessor}})
-            {{/HasBody}}
-            .send()
-            .await
-            .map_err(Error::io)?;
-        if !res.status().is_success() {
-            let status = res.status().as_u16();
-            let headers = gax::error::convert_headers(res.headers());
-            let body = res.bytes().await.map_err(Error::io)?;
+        self.execute(builder, {{#HasBody}}Some(req{{BodyAccessor}}){{/HasBody}}{{^HasBody}}None::<NoBody>{{/HasBody}}).await
+    }
+    {{/Methods}}
+
+    async fn execute<I: serde::ser::Serialize, O: serde::de::DeserializeOwned>(
```

Comment:
For a future PR: how do we deal with services that have an `execute()` RPC?

Reply:
ack, cross that when we get there. I am sure we can mangle or throw an underscore in front or something.